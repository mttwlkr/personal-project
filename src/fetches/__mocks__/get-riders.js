export const getRiders = jest.fn()