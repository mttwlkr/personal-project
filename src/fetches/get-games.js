export const getGames = async () => {
  try {
    const proxyurl = "https://cors-anywhere.herokuapp.com/";    
    const url = "slvsh.com/posts.json?page=1&is_active=true";
    const response = await fetch(proxyurl + url)
    const games = await response.json()
    return games
  } catch (error) {
    throw await error 
  }    
}