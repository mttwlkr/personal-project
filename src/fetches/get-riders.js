export const getRiders = async () => {
  try {
    const proxyurl = "https://cors-anywhere.herokuapp.com/";    
    const url = "slvsh.com/riders.json";
    const response = await fetch(proxyurl + url);
    const riders = await response.json();
    return riders;
  } catch (error) {
    throw await error;
  }      
};