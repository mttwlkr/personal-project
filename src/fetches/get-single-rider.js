export const getSingleRider = async (riderID) => {
  try {
    const proxyurl = "https://cors-anywhere.herokuapp.com/";
    const url = `slvsh.com/riders/${riderID}.json`
    const response = await fetch(proxyurl + url)
    const rider = await response.json()
    return rider
  } catch (error) {
    throw error
  }
}