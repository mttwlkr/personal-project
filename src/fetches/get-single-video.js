export const getSingleVideo = async (gameID) => {
  try {
    const proxyurl = "https://cors-anywhere.herokuapp.com/"; 
    const response = await fetch(`${proxyurl}slvsh.com/${gameID}.json`)
    const game = await response.json()
    return game
  } catch (error) {
    throw error
  }
}
