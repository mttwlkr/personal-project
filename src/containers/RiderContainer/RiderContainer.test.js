import React from 'react';
import { shallow } from 'enzyme';
import { RiderContainer, 
  mapStateToProps, 
  mapDispatchToProps }  from './RiderContainer.js';

import { getRiders } from '../../fetches/get-riders';
import { mockAPIRidersData, 
  mockHistory } from '../../mock-data/mockData';

jest.mock('../../fetches/get-riders');

describe('RiderContainer', () => {

  const mockAddStatsToStore = jest.fn();
  const mockAddRidersToStore = jest.fn();
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<RiderContainer 
      addRidersToStore={mockAddRidersToStore}
      addStatsToStore={mockAddStatsToStore}
      riders={mockAPIRidersData}
      history={mockHistory}
    />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should invoke getRiders and addRidersToStore on load', () => {
    const mockAddRidersToStore = jest.fn();
    const mockEmptyRiders = [];
    
    wrapper = shallow(<RiderContainer 
      addRidersToStore={mockAddRidersToStore}
      addStatsToStore={mockAddStatsToStore}
      riders={mockEmptyRiders}
      history={mockHistory}
    />);
    expect(getRiders).toHaveBeenCalled();
  });

  it('handleOriginal should set state', () => {
    let mockEvent = {target: {className: 'rt-th slvsh-stats-header'}};
    wrapper.instance().handleOriginal(mockEvent);
    expect(wrapper.state('clicked')).toEqual(true);
  });

  it('handleOriginal should set state', () => {
    let mockEvent = {target: {className: 'trick-key-div'}};
    wrapper.instance().handleOriginal(mockEvent);
    expect(wrapper.state('clicked')).toEqual(true);
  });

  it('handleOriginal should set state', () => {
    let mockEvent = {target: {className: 'trick-key'}};
    wrapper.instance().handleOriginal(mockEvent);
    expect(wrapper.state('clicked')).toEqual(true);
  });

  it('handleRider should update history', () => {
    wrapper.instance().handleRider(23);
    expect(wrapper).toMatchSnapshot();
  });

  it('handleOriginal should set state to false if not correct class', () => { 
    let mockEvent = {target: {className: 'something-that-doesnt-match'}};
    wrapper.instance().handleOriginal(mockEvent);
    expect(wrapper.state('clicked')).toEqual(false);
  });

  it.skip('should invoke handleOriginal on click', () => {
    const spy = jest.spyOn(wrapper.instance(), 'handleOriginal');
    wrapper.find('rt-th slvsh-stats-header').simulate('click');
    expect(spy).toHaveBeenCalled();
  });

  it('should mapStateToProps correctly', () => {
    const mockState = {riders: [{name: 'Jonny Moesley', id: 30}]};
    const mapped = mapStateToProps(mockState);
    expect(mapped.riders).toEqual(mockState.riders);
  });

  it('should mapDispatchToProps correctly', () => {
    const mockDispatch = jest.fn();
    const mapped = mapDispatchToProps(mockDispatch);
    mapped.addRidersToStore();
    mapped.addStatsToStore();
    expect(mockDispatch).toHaveBeenCalled();
  });
});