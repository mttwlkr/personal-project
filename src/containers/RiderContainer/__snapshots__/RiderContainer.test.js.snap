// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RiderContainer should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RiderContainer
    addRidersToStore={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "I'm a rider container",
      "className": "rider-container",
    },
    "ref": null,
    "rendered": "I'm a rider container",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "I'm a rider container",
        "className": "rider-container",
      },
      "ref": null,
      "rendered": "I'm a rider container",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
